src/test/java/org/broadinstitute/hellbender/tools/walkers/vqsr/scalable/ExtractVariantAnnotationsIntegrationTest.java

```diff
@@ -0,0 +1,253 @@
+                Arrays.asList(
+                        Pair.of("snp", ADD_SNP_MODE_AND_RESOURCES),
+                        Pair.of("indel", ADD_INDEL_MODE_AND_RESOURCES),
+                        Pair.of("snpIndel", ADD_SNP_MODE_AND_RESOURCES.andThen(ADD_INDEL_MODE_AND_RESOURCES))),
+                Arrays.asList(
+                        Pair.of("pos", Function.identity()),
+                        Pair.of("posUn", ADD_MAXIMUM_NUMBER_OF_UNLABELED_VARIANTS)));
+
+        return testConfigurations.stream()
+                .map(tagAndAddFunctionPairs -> new Object[]{
+                        tagAndAddFunctionPairs.stream().map(Pair::getLeft).collect(Collectors.joining(".")),    // e.g., "extract.nonAS.snp.pos"
+                        tagAndAddFunctionPairs.stream().map(Pair::getRight)                                              // creates the corresponding ArgumentsBuilder
+                                .reduce(Function.identity(), Function::andThen)                                          //  by stringing together functions that add the
+                                .apply(BASE_ARGUMENTS_BUILDER_SUPPLIER.get())})                                          //  appropriate arguments
+                .toArray(Object[][]::new);
+    }
+
+    /**
+     * Checks expected outputs given a tag (e.g., "extract.nonAS.snp.pos") and arguments corresponding to the
+     * Cartesian products generated in {@link #dataValidInputs}.
+     *
+     * We perform exact-match tests of any annotation HDF5 files produced using h5diff, which is insensitive to timestamps within the file.
+     * We also perform exact-match tests of VCF files using diff. VCF indices may not be diff equivalent, so
+     * we just check for their existence.
+     */
+    @Test(dataProvider = "dataValidInputs", groups = {"python"}) // python environment is required to use h5diff for exact-match comparisons
+    public void testValidInputs(final String tag,
+                                final ArgumentsBuilder argsBuilder) {
+        final File outputDir = UPDATE_EXACT_MATCH_EXPECTED_OUTPUTS ? EXPECTED_TEST_FILES_DIR : createTempDir("extract");
+        final String outputPrefix = String.format("%s/%s", outputDir, tag);
+        argsBuilder.addOutput(outputPrefix);
+        runCommandLine(argsBuilder);
+
+        if (!UPDATE_EXACT_MATCH_EXPECTED_OUTPUTS) {
+            assertOutputs(tag, outputPrefix);
+        }
+    }
+
+    private static void assertOutputs(final String tag,
+                                      final String outputPrefix) {
+        // vcf.idx files are not reproducible
+        SystemCommandUtilsTest.runSystemCommand(String.format("h5diff %s/%s %s",
+                EXPECTED_TEST_FILES_DIR,
+                tag + LabeledVariantAnnotationsWalker.ANNOTATIONS_HDF5_SUFFIX,
+                outputPrefix + LabeledVariantAnnotationsWalker.ANNOTATIONS_HDF5_SUFFIX));
+        SystemCommandUtilsTest.runSystemCommand(String.format("diff %s/%s.vcf %s.vcf",
```

Comment:
I'm surprised you don't need to strip out the header from these files before doing the diff. Does that mean that none of these tools add a program line to their output VCFs? They probably should if they aren't currently.

Reply:
Oops, good catch. I'll use `getDefaultToolVCFHeaderLines` but then set `addOutputVCFCommandLine` to false for all tests, so long as we're OK with not testing for the presence of these lines.

Reply:
Haha actually wait, this is already how it is!

Reply:
Oops sorry, totally missed that.